analyzers/src/SonarAnalyzer.Common/Helpers/NamingHelper.cs

```diff
@@ -32,6 +32,6 @@ public static class NamingHelper
         internal const string CamelCasingPatternWithOptionalPrefixes = "^(s_|_)?" + CamelCasingInternalPattern + "$";
 
         internal static bool IsRegexMatch(string name, string pattern) =>
-            Regex.IsMatch(name, pattern, RegexOptions.CultureInvariant | RegexOptions.Compiled, RegexConstants.DefaultTimeout);
+            RegexExtensions.SafeIsMatch(name, pattern, RegexOptions.CultureInvariant | RegexOptions.Compiled);
```

Comment:
Any problem with this being a normal method instead of an extension method?

IMO it's fine to put it in `RegexExtensions`, since it thematically fits.
Doing `this string input` or `this string pattern` does not make much sense to me.

Reply:
I agree. It's easier for me to read after `Regex.IsMatch` -> `RegexExtensions.SafeIsMatch` than an extension method on `string`.